src/plays/web3-staking-dapp/Web3StakingDapp.js

```diff
@@ -0,0 +1,273 @@
+import PlayHeader from 'common/playlists/PlayHeader';
+import './styles.css';
+import React, { useState, useEffect } from "react";
+import artifact from "./artifacts/contracts/Staking.sol/Staking.json";
+import { ethers } from "ethers";
+import Navbar from "./components/Navbar";
+import maticLogo from "./images/polygon-matic-logo.svg";
+import { FaRupeeSign } from 'react-icons/fa';
+
+// WARNING: Do not change the entry componenet name
+function Web3StakingDapp(props) {
+
+  // Your Code Start below.
+
+  const ContractAddress = '0x12163B070B97f06F5061D93164D960bbFCfdf965';
+
+  // frontend states
+  const [provider, setProvider] = useState(undefined);
+  const [account, setAccount] = useState(undefined);
+  const [contract, setContract] = useState(undefined);
+  const [accountAddress, setAccountAddress] = useState(undefined);
+
+  // assets
+  const [assetId, setAssetId] = useState([]);
+  const [assets, setAssets] = useState([]);
+
+  // staking
+  const [stakeModal, setStakeModal] = useState(false);
+  const [stakingLength, setStakingLength] = useState(undefined);
+  const [stakingPercent, setStakingPercent] = useState(undefined);
+  const [amount, setAmount] = useState(0);
+
+  // functions
+  const toWei = ether => ethers.utils.parseEther(ether);
+  const toMatic = wei => ethers.utils.formatEther(wei);
+
+  useEffect(() => {
+    const onload = async () => {
+      const provider = new ethers.providers.Web3Provider(window.ethereum);
+      setProvider(provider);
+
+      const contract = new ethers.Contract(
+        ContractAddress,
+        artifact.abi
+      )
+      setContract(contract);
+    }
```

Comment:
Can we Move this function out of the useEffect?

Reply:
then it will show a warning i think

Reply:
Which warning?